src/client/components/ui/TipDisclosure/TipDisclosure.test.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+
+import { KeyBoard } from '~/client/components/keyboard.fixture';
+import { TipDisclosure } from '~/client/components/ui/TipDisclosure/TipDisclosure';
+
+describe('TipDisclosure', () => {
+	it.each([KeyBoard.ENTER, KeyBoard.SPACE])
+	('il est possible d’ouvrir et fermer l’infobulle avec la touche Enter et Space', async (touche: KeyBoard) => {
+		// Given
+		const user = userEvent.setup();
+		const ariaLabel = 'buttonLabel';
+		render(
+			<TipDisclosure icon="information" disclosureAriaLabel={ariaLabel} tipId='test'>ceci est un texte informatif</TipDisclosure>,
+		);
+		await user.tab();
+
+		// When
+		const boutonDisclosure = screen.getByRole('button', { name: 'buttonLabel (Ouvrir)' });
+		expect(boutonDisclosure).toHaveFocus();
+		await user.keyboard(touche);
+
+		// Then
+		const information = screen.getByText('ceci est un texte informatif');
+
+		expect(information).toBeVisible();
+		expect(boutonDisclosure).toHaveAttribute('aria-expanded', 'true');
+		expect(boutonDisclosure).toHaveAttribute('aria-controls', 'test');
+		expect(boutonDisclosure).toHaveAccessibleName('buttonLabel (Fermer)');
+
+
+		// When
+		await user.keyboard(touche);
+
+		//Then
+		expect(information).not.toBeVisible();
```

Comment:
toBeVisible ou to beIntheDocument ?

Reply:
pourquoi utiliser hidden et pas &&

Reply:
J'ai mis plutôt `toBeVisible` que `toBeInTheDocument` car si il est absent du DOM / présent dans le DOM mais caché avec `hidden` ou `display:none`, ça me semblait pareil

Reply:
Et le `hidden` -> c'était le code qui était présent avant 
Ça m'a pas fait tiquer, il y a une préférence pour les `&&` ?